indico/modules/users/client/js/react/components/ICSCalendarLink.jsx

```diff
@@ -0,0 +1,135 @@
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import {Button, Dropdown, Icon, Input, Label, Grid, Popup, Header} from 'semantic-ui-react';
+
+import signURL from 'indico-url:core.sign_url';
+
+import {Translate} from 'indico/react/i18n';
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+import {snakifyKeys} from 'indico/utils/case';
+
+export default function ICSCalendarLink({endpoint, urlParams, options, ...restProps}) {
+  const [copied, setCopied] = useState(false);
+  const [option, setOption] = useState(null);
+  const [open, setOpen] = useState(false);
+
+  const copyButton = (
+    <Button
+      icon="copy"
+      onClick={async () => {
+        await navigator.clipboard.writeText(option.url);
+        setCopied(true);
+      }}
+    />
+  );
+
+  const fetchURL = async queryParams => {
+    try {
+      const {
+        data: {url: signedURL},
+      } = await indicoAxios.post(signURL(), snakifyKeys({endpoint, urlParams, queryParams}));
+      return signedURL;
+    } catch (error) {
+      handleAxiosError(error);
+    }
+  };
+
+  const trigger = (
+    <Dropdown
+      {...restProps}
+      icon={null}
+      trigger={
+        <Button icon size="small">
+          <Icon name="calendar alternate outline" />
+          <Icon name="caret down" />
+        </Button>
+      }
+      pointing="top right"
+    >
+      <Dropdown.Menu>
+        <Dropdown.Header>
+          <Translate>Synchronize with your calendar</Translate>
```

Comment:
`Synchronise` :gb: ?

(at some point I'd like to move all source strings to en_US to be consistent with non-user-facing code and just use transifex for the en_GB-isms)